PolicyServicePkg/Library/VerifiedPolicy.c

```diff
@@ -1,247 +1,247 @@
-  IN EFI_HANDLE  DataHandle
-  )
-{
-  if (DataHandle == NULL) {
-    return EFI_INVALID_PARAMETER;
-  }
-
-  FreePool (DataHandle);
-  return EFI_SUCCESS;
-}
-
-/**
-  Records access to a policy data structure used by autogenerated code. This
-  function should not be manually called.
-
-  @param[in]    DataHandle      The policy handle where the access was made.
-  @param[in]    CallerGuid      The file guid of the caller for tracking access.
-  @param[in]    Offset          The offset into the policy data for the access.
-  @param[in]    Size            The size of the access.
-  @param[in]    Write           Indicates if the policy was written to.
-
-**/
-VOID
-EFIAPI
-ReportVerifiedPolicyAccess (
-  IN EFI_HANDLE      DataHandle,
-  IN CONST EFI_GUID  *CallerGuid,
-  IN UINT32          Offset,
-  IN UINT32          Size,
-  IN BOOLEAN         Write
-  )
-
-{
-  //
-  // Currently a no-op, but leaving room for future checks and analytics.
-  //
-
-  return;
-}
+/** @file
+  The common implementation of the verified policy library routines.
+
+  Copyright (c) Microsoft Corporation
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+#include <Library/UefiLib.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/MemoryAllocationLib.h>
+
+#include <Library/VerifiedPolicy.h>
+#include <PolicyInterface.h>
+
+#include "../PolicyLibCommon.h"
+
+/**
+  Retrieves a verified policy of the given type from the policy store.
+
+  @param[in]    PolicyGuid      The GUID of policy in the policy store.
+
+  @param[in]    Descriptor      The descriptor for the verified policy data
+                                structure.
+
+  @param[out]   Attributes      Returns the attributes of the policy in the
+                                policy store.
+
+  @param[out]   DataHandle      Returns the handle to the verified policy.
+
+  @retval   EFI_SUCCESS               The policy was successfully retrieved.
+  @retval   EFI_BAD_BUFFER_SIZE       The policy was an unexpected size.
+  @retval   EFI_INCOMPATIBLE_VERSION  The verified policy major version did not
+                                      match.
+  @retval   EFI_OUT_OF_RESOURCES      Failed to allocate memory.
+**/
+RETURN_STATUS
+EFIAPI
+GetVerifiedPolicy (
+  IN CONST EFI_GUID                    *PolicyGuid,
+  IN CONST VERIFIED_POLICY_DESCRIPTOR  *Descriptor,
+  OUT UINT64                           *Attributes OPTIONAL,
+  OUT EFI_HANDLE                       *DataHandle
+  )
+{
+  VOID                    *Buffer;
+  UINT16                  BufferSize;
+  POLICY_INTERFACE        *PolicyService;
+  EFI_STATUS              Status;
+  VERIFIED_POLICY_HEADER  *Header;
+
+  Buffer      = NULL;
+  BufferSize  = 0;
+  Header      = NULL;
+  *DataHandle = NULL;
+
+  Status = GetPolicyInterface (&PolicyService);
+  if (EFI_ERROR (Status)) {
+    goto Exit;
+  }
+
+  Status = PolicyService->GetPolicy (PolicyGuid, NULL, Attributes, &Buffer, &BufferSize);
```

Comment:
VerifiedPolicy.c's APIs should be updated to include Name as a passed parameter, right?

Reply:
I actually intentionally left these as is. The verified policies will all be auto-generated and therefor should have a unique GUID with no name. Do we intend to use the verified policy lib for the config work?

Reply:
That's how I have it written up currently, exclusively using the verified policy lib. My understanding was that was the preferred mechanism to interact with policies. Have I misunderstood? I don't see where the auto-generation is happening, the caller of SetVerifiedPolicy passes the GUID that is used, right? My thought was that the caller would also pass the Name, if applicable.